src/main/java/io/jenkins/plugins/audit/listeners/BuildStartListner.java

```diff
@@ -0,0 +1,42 @@
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.listeners.RunListener;
+import org.apache.logging.log4j.audit.LogEventFactory;
+import io.jenkins.plugins.audit.event.BuildStart;
+
+import java.util.Date;
+
+
+@Extension
+public class BuildStartListner extends RunListener<Run> {
+    /**
+     * Fired when a build is started, event logged via Log4j-audit.
+     *
+     * @param run of type Run having the build information
+     * @param listener of type TaskListener that the onStarted method expects
+     */
+    @Override
+    public void onStarted(Run run, TaskListener listener) {
+        BuildStart buildStart = LogEventFactory.getEvent(BuildStart.class);
+
+        buildStart.setBuildNumber(run.getNumber());
+        //run.getCause(<?>).getShortDescription(); Not sure how to get the cause.
```

Comment:
Can you please help me use this?
https://javadoc.jenkins.io/hudson/model/Run.html#getCause

Reply:
attached here some code i had used earlier for logging build starts. you might find it useful-
https://gist.github.com/davidolorundare/3a7fdf8cc964aaa4ab9b7f2baf237f32

Reply:
You'll want to use `run.getCauses()` instead. I think you could adjust the audit events to take a list of causes instead of just the one.

Reply:
Thanks! I hope this is resolved.